mgmtsystem/virtualcenter.py

```diff
@@ -572,30 +655,31 @@ def clone_vm(self, source, destination, resourcepool=None, datastore=None, power
             folder = source_template.parent.parent.vmParent
         except AttributeError:
             folder = source_template.parent
-
         progress_callback("Picked folder `{}`".format(folder.name))
 
         task = source_template.CloneVM_Task(folder=folder, name=destination, spec=vm_clone_spec)
-
-        def _check():
+        start_time = time.time()
+        delta = 0
+        while task.info.state not in {'queued', 'running'} and delta < provision_timeout:
```

Comment:
@RonnyPfannschmidt These objects are pyVmomi objects (not my own), they directly correspond against the objects in the vSphere API, and they are read-only. You get an exception if you try to update any of the properties on the objects. I basically do a query to fetch that particular object again, and return the new instance.